src/bid_order.rs

```diff
@@ -216,42 +150,6 @@ pub fn migrate_bid_orders(
     let version_info = get_version_info(store)?;
     let current_version = Version::parse(&version_info.version)?;
 
-    // version support added in 0.15.0, all previous versions migrate to v1 of state data
-    let upgrade_req = VersionReq::parse("<0.15.0")?;
-
-    if upgrade_req.matches(&current_version) {
-        let existing_bid_order_ids: Vec<Vec<u8>> = bucket_read(store, NAMESPACE_ORDER_BID)
-            .range(None, None, Order::Ascending)
-            .map(|kv_bid: StdResult<Record<BidOrder>>| {
-                let (bid_key, _) = kv_bid.unwrap();
-                bid_key
-            })
-            .collect();
-
-        for existing_bid_order_id in existing_bid_order_ids {
-            let existing_bid_order: BidOrder =
-                bucket_read(store, NAMESPACE_ORDER_BID).load(&existing_bid_order_id)?;
-            get_bid_storage(store).save(&existing_bid_order_id, &existing_bid_order.into())?
-        }
-    }
-
-    // migration from 0.15.2 - 0.16.1 => latest
-    if VersionReq::parse(">=0.15.1, <0.16.2")?.matches(&current_version) {
-        let existing_bid_order_ids: Vec<Vec<u8>> = bucket_read(store, NAMESPACE_ORDER_BID)
-            .range(None, None, Order::Ascending)
-            .map(|kv_bid: StdResult<Record<BidOrderV1>>| {
-                let (bid_key, _) = kv_bid.unwrap();
-                bid_key
-            })
-            .collect();
-
-        for existing_bid_order_id in existing_bid_order_ids {
-            let existing_bid_order: BidOrderV1 =
-                bucket_read(store, NAMESPACE_ORDER_BID).load(&existing_bid_order_id)?;
-            get_bid_storage(store).save(&existing_bid_order_id, &existing_bid_order.into())?
-        }
-    }
-
     // bid fees and order events added in 0.16.3, refunds may be necessary for existing orders
     if VersionReq::parse("<0.16.3")?.matches(&current_version) {
```

Comment:
Is this block still needed @brashidnia-figure ?

Reply:
```
if VersionReq::parse("<0.16.3")?.matches(&current_version) {
   return Err(ContractError:: Some error)
}
```

i believe this check will return an error and fail the migration if the contract version is older than v0.16.3. This will allow removal of `BidOrderV1` and `BidOrder`


then for the new stuff (BidOrderV3) 

```
if VersionReq::parse("<0.18.1")?.matches(&current_version) {
  // logic to migrate bid orders with events + fees to new data structure in your other PR
}
```

which will basically upgrade `v0.16.3` through `v0.18.0` to the new stuff in your other PR

comment here explaning https://github.com/FigureTechnologies/ats-smart-contract/pull/110#discussion_r1174089869

Reply:
I'll add `BidOrderV3` + the check for `if VersionReq::parse("<0.18.1")?.matches(&current_version) { ... }` to #110 after this is finished and merged.